grpcgcp/gcp_balancer.go

```diff
@@ -53,6 +53,7 @@ func (bb *gcpBalancerBuilder) Build(
 	return &gcpBalancer{
 		cc:          cc,
 		affinityMap: make(map[string]balancer.SubConn),
+		fallbackMap: make(map[string]balancer.SubConn),
```

Comment:
almost everything is serialized in balancer by mutex. Is this fine because balancer should not be called that often?

Reply:
Right. The Picker is called on every RPC call but the balancer only when DNS changes or subconn state changes.

Reply:
In our case the Picker also calls Balancer's `getReadySubConnRef` which is synchronized. But I am going to re-introduce a lock in picker in the following PRs to make sure we are not exceeding stream limit. And according to some benchmarks I ran, without synchronization it gives ~0.7us +-20-30% vs with synchronization ~0.9us+-1-3% depending on number of cores and concurrent  calls.

Reply:
If it is not over optimization we could use atomics.